### PR TITLE
Add possibility of using Wireshark in CRTP debugging

### PR DESCRIPTION
**Pull request by jonasdn**
# cflib: Add PCAP log

After this you can specify:

```
$ CRTP_PCAP_LOG=filename.pcap python3 examples/swarm/hl-commander-swarm.py
```

To generate a PCAP file with the CRTP communication seen by crazyradio.

The PCAP record generated is structured as:

| link type | receive | address | channel | radio devid | crtp header | crtp data |
| ----------- | ----------| ------------| ---------- | --------------- | --------------- | -----------  |
| 1 byte    | 1 byte   | 5 bytes  |  1 byte   |    1 byte      |    1 byte      |   n bytes |

Where the receive bye indicates whether this is an incoming packet or not.

# Add Wireshark dissector

fancy

```bash
$ tshark -r test.pcap "crtp.port == 8"
    6   0.003333 00:DE:AD:BE:EF (10) → Radio #0     CRTP 13 Setpoint Highlevel
13158   5.176752     Radio #0 → 00:DE:AD:BE:EF (10) CRTP 24 Setpoint Highlevel
13163   5.178626 00:DE:AD:BE:EF (10) → Radio #0     CRTP 13 Setpoint Highlevel
17768   8.179085     Radio #0 → 00:DE:AD:BE:EF (10) CRTP 32 Setpoint Highlevel
17773   8.181294 00:DE:AD:BE:EF (10) → Radio #0     CRTP 13 Setpoint Highlevel
21223  10.181543     Radio #0 → 00:DE:AD:BE:EF (10) CRTP 32 Setpoint Highlevel
21228  10.183293 00:DE:AD:BE:EF (10) → Radio #0     CRTP 13 Setpoint Highlevel
24030  12.182588     Radio #0 → 00:DE:AD:BE:EF (10) CRTP 32 Setpoint Highlevel
24035  12.185723 00:DE:AD:BE:EF (10) → Radio #0     CRTP 13 Setpoint Highlevel
26449  14.184183     Radio #0 → 00:DE:AD:BE:EF (10) CRTP 32 Setpoint Highlevel
26454  14.185862 00:DE:AD:BE:EF (10) → Radio #0     CRTP 13 Setpoint Highlevel
28388  16.187258     Radio #0 → 00:DE:AD:BE:EF (10) CRTP 24 Setpoint Highlevel
28393  16.188963 00:DE:AD:BE:EF (10) → Radio #0     CRTP 13 Setpoint Highlevel
30533  18.190823     Radio #0 → 00:DE:AD:BE:EF (10) CRTP 11 Setpoint Highlevel
```

### CODE REVIEW

**Comment by jonasdn** [comment]
So, @ataffanel you see I adder some general (non-CRTP) Crazyradio data to the PCAP. Like the source address and the devid of the radio and also a byte indicating if the packet was incoming or not. 

This is not well thought out, neither in regard to the right now nor in regard to a shiny new future. And we could probably do better and achieve the same. Any ideas?

Could we encode something more than the devid in a byte?

Like:

| Bit 0 - 4 | Bit 5 - 8 |
| ---------- | ----------- |
| usb/radio/ble/other| identifier |

Do we need a byte indicating incoming packet or can we infer it in someway?

File: wireshark/README.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
+# How to analyze CFLIB CRTP traffic using Wireshark
```
Comment (knmcguire) [inline wireshark/README.md:1]: Shouldn't this be part of the CFlib doc folder? it would be good to have all documentation at on place.

**Comment by ataffanel** [comment]
Looks great!

The dissector file should likely not be in the project root. I think the "tools" folder would feel better an less noisy.

The decoding is a bit "sparse" so far. This is something that can be improved later but it would be nice to add at least an "undecoded" field to show the user the rest of the packet, for example in that case to make clear that there is data in the packet:
Screenshot from 2022-01-12 12-59-45

**Review by ataffanel — CHANGES_REQUESTED** [review]
See my comment.

**Comment by jonasdn** [comment]
Thanks for comments! I pushed two commits addressing your issues!

**Review by ataffanel — APPROVED** [review]

**Comment by ataffanel** [comment]
Looks great to me, great addition!

**Pull request merged**